sdk/src/main/java/com/bugsnag/android/SessionTracker.java

```diff
@@ -78,6 +81,29 @@
         return session;
     }
 
+    void startSession(boolean autoCaptured) {
+        startNewSession(new Date(), client.getUser(), autoCaptured);
+    }
+
+    void stopSession() {
+        Session session = currentSession.get();
+
+        if (session != null) {
+            session.isStopped.set(true);
+        }
+    }
+
+    boolean resumeSession() {
+        Session session = currentSession.get();
+
+        if (session == null) {
+            startSession(false);
+            return false;
+        } else {
+            return session.isStopped.compareAndSet(true, false);
```

Comment:
Does this cope with the case "The previous session ended in a crash. In other words, one session can only have 1 crash at most." in the project definition? Should we also check if there have been unhandled errors?

Reply:
I think @duncanhewett said we don't need to enforce this now.